tensorflow_model_optimization/python/core/sparsity/keras/pruning_impl.py

```diff
@@ -38,13 +38,17 @@ def __init__(self, training_step_fn, pruning_vars, pruning_schedule,
         throughout training.
       block_size: The dimensions (height, weight) for the block sparse pattern
         in rank-2 weight tensors.
+      sparsity_2x4: Boolean that indicates whether we should do sparsity_2x4.
+        We check whether we can do pruning with 2x4 sparsity. If not, then
+        we fallback to the default unstructured pruning for 50%.
```

Comment:
I believe raising error when 2:4 sparsity fails, rather than falling back to unstructured pruning. 
Since user explicitly passed the structured sparsity flag to the API, it is intuitive to fail when it is not possible. Fallback can confuse users who expect structured sparsity but get unstructured one.

Reply:
Done.